src/malli/core.cljc

```diff
@@ -177,7 +177,16 @@
 (defn -set-entries [schema key value]
   (let [found (atom nil)
         [key :as new-child] (if (vector? key) (conj key value) [key value])
-        children (cond-> (mapv (fn [[k :as child]] (if (= key k) (do (reset! found true) new-child) child)) (-children schema))
+        new-props (when (= (count new-child) 3) (nth new-child 1))
+        children (cond-> (mapv (fn [[k props :as child]]
```

Comment:
what about format `[:user/id {:optional true}]|`? are the entries passed in as tuple-3 here? If, no need to check the length, if not, this doesn't work correctly

Reply:
Simplified the check a bit